benchmark/scripts/Benchmark_Driver

```diff
@@ -436,122 +373,95 @@ class BenchmarkDoctor(object):
-            )
-            BenchmarkDoctor.log_naming.info("See http://bit.ly/BenchmarkNaming")
+                "'%s' name doesn't conform to benchmark naming convention.",
+                name)
+            BenchmarkDoctor.log_naming.info(
+                'See http://bit.ly/BenchmarkNaming')
 
         if composite_words > 4:
             BenchmarkDoctor.log_naming.warning(
-                "'%s' name is composed of %d words.", name, composite_words
-            )
+                "'%s' name is composed of %d words.", name, composite_words)
             BenchmarkDoctor.log_naming.info(
                 "Split '%s' name into dot-separated groups and variants. "
-                "See http://bit.ly/BenchmarkNaming",
-                name,
-            )
+                "See http://bit.ly/BenchmarkNaming", name)
 
     @staticmethod
     def _name_is_at_most_40_chars_long(measurements):
-        name = measurements["name"]
+        name = measurements['name']
 
         if len(name) > 40:
             BenchmarkDoctor.log_naming.error(
-                "'%s' name is %d characters long.", name, len(name)
-            )
+                "'%s' name is %d characters long.", name, len(name))
             BenchmarkDoctor.log_naming.info(
-                "Benchmark name should not be longer than 40 characters."
-            )
+                'Benchmark name should not be longer than 40 characters.')
 
     @staticmethod
-    def _select(measurements, num_iters=None, opt_level="O"):
-        prefix = measurements["name"] + " " + opt_level
-        prefix += "" if num_iters is None else (" i" + str(num_iters))
-        return [
-            series for name, series in measurements.items() if name.startswith(prefix)
-        ]
+    def _select(measurements, num_iters=None, opt_level='O'):
+        prefix = measurements['name'] + ' ' + opt_level
+        prefix += '' if num_iters is None else (' i' + str(num_iters))
+        return [series for name, series in measurements.items()
+                if name.startswith(prefix)]
 
     @staticmethod
     def _optimized_runtime_in_range(measurements):
-        name = measurements["name"]
+        name = measurements['name']
         setup, ratio = BenchmarkDoctor._setup_overhead(measurements)
         setup = 0 if ratio < 0.05 else setup
         runtime = min(
-            [
-                (result.samples.min - correction)
-                for i_series in [
-                    BenchmarkDoctor._select(measurements, num_iters=i)
-                    for correction in [(setup / i) for i in [1, 2]]
-                ]
-                for result in i_series
-            ]
-        )
+            [(result.samples.min - correction) for correction, i_series in
+             [(correction, BenchmarkDoctor._select(measurements, num_iters=i))
+              for i, correction in [(i, setup // i) for i in [1, 2]]]
+             for result in i_series])
```

Comment:
I personally find deeply nested list comprehensions significantly harder to read compared to for loops. Especially here, where you have to propagate the `i` and `correction` variables, which would not be necessary if we were using the for loop:

```python
runtimes = []
for i in range(1, 3):
    correction = setup / i
    i_series = BenchmarkDoctor._select(measurements, num_iters=i)
    for result in i_series:
        runtimes.append(result.samples.min - correction)
runtime = min(runtimes)
```

Reply:
This really comes down to personal taste. I have trouble reading procedural code with side effects. I’ve [answered below](https://github.com/apple/swift/pull/30158#commitcomment-37585480) with link to direct comparison of the Python 3 changes.

Reply:
@eeckstein Do you have a preference here?

Reply:
I find @broadwaylamb's version _much_ easier to understand. That's probably because I'm not a python expert (which is probably the case for many our contributors).

In general, making the code easy to understand should be a central goal.

Reply:
Refactored in bf06df6.